### ISSUE TITLE
Support for credentials via AWS SSO

### ISSUE THREAD

**Issue by foxylion** [aws/aws-sdk-js-v3#1158]
**Is your feature request related to a problem? Please describe.**
The new AWS CLI v2 supports a profile configuration using [SSO credentials](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-sso.html).
This improves security as there are no longer any long-term credentials (access-key-id, secret-access-key) stored on the device.

**Describe the solution you'd like**
AWS SDK for Javascript (v3) should add support for SSO profiles in `~/.aws/config`.

**Describe alternatives you've considered**
Current workaround is to copy & paste the credentials as environment variables provided by the AWS SSO sign-in page.

**Comment by villelahdenvuo** [comment]
This is very much needed! I made a hacky script that figures out the current CLI session from .aws/cli/cache and sets the environment variables automatically, but I shouldn't need to do that.

**Comment by benkehoe** [comment]
I made a tool that uses the `credential_process` mechanism to add support to SDKs that don't support AWS SSO yet: [aws-sso-util](https://github.com/benkehoe/aws-sso-util#adding-aws-sso-support-to-aws-sdks), which supports refreshing credentials so you don't run into credential expiration, etc.

**Comment by monken** [comment]
Another workaround for bash and zsh shells. Add this to your ~/.bashrc or ~/.zshrc.

Usage: `awscreds myprofile`. This will export the credentials into the current terminal session or trigger the sign in process if the credentials expired.

```bash
function awscreds() {
    export AWS_PAGER=""
    aws sts get-caller-identity --profile $1 || aws sso login --profile $1
    FILE=~/.aws/cli/cache/$(ls -t ~/.aws/cli/cache | head -n 1)

    export AWS_ACCESS_KEY_ID="$(jq -r '.Credentials.AccessKeyId' $FILE)" AWS_SECRET_ACCESS_KEY="$(jq -r '.Credentials.SecretAccessKey' $FILE)" AWS_SESSION_TOKEN="$(jq -r '.Credentials.SessionToken' $FILE)"
}
```

**Comment by benkehoe** [comment]
@monken I've also got [`aws-export-credentials`](https://github.com/benkehoe/aws-export-credentials) to make the process of retrieving and injecting credentials simpler and more reliable. You'd do
```
function awscreds() {
    AWS_PAGER="" aws sts get-caller-identity --profile $1 || AWS_PAGER="" aws sso login --profile $1
    export "$(aws-export-credentials --env --profile $1)"
}
```
separately, your `export AWS_PAGER=""` would affect subsequent CLI commands, which I don't think is your intent.

**Comment by urz9999** [comment]
In case like this one or other similar cases where AWS SSO result in incompatibilities with your library and you don't want to play with workarounds or complicated fixes, maybe you can give a try to our open-source project: [https://github.com/Noovolari/leapp](https://github.com/Noovolari/leapp). It deals with AWS SSO authentication and accounts/roles retrieval then it creates short-lived temporary credentials in .aws/credentials to maximize compatibility with third party tools / sdks.

**Comment by benkehoe** [comment]
Is this accessible through npm yet? If not, can you re-open this and close it once a version has been released?

**Comment by AllanZhengYP** [comment]
Hi @benkehoe 

The SSO credential provider has been released to NPM since  [3.7.0](https://github.com/aws/aws-sdk-js-v3/blob/main/CHANGELOG.md#370-2021-02-25). Here's the documentation with example: https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/modules/_aws_sdk_credential_provider_sso.html
